---
title: "Nanopore Processing"
output: html_notebook
---
```{r}
library(tidyverse)
library(ggsci)
library(ggpubr)
library(biomaRt)
library(refGenome)
```


Notebook Theme:
```{r}
this_theme <- theme(text = element_text(size=12, family = "sans"),
      legend.title = element_text(size = 20),
      axis.text.x = element_text(color = 'black',
                                 size = 15, face = 'bold'),
      axis.text.y = element_text(color = 'black', 
                                 size=15, face = 'bold'),
      panel.background = element_rect(fill ='white'),
      axis.line.x = element_line(size=1),
      axis.line.y = element_line(size=1),
      strip.background = element_blank(),
      strip.text = element_text(face = 'bold', 
                                size = 5, hjust=0.95),
      legend.position='top',
      legend.key = element_blank(),
      axis.ticks.y = element_blank(), 
      axis.title.y = element_text(face='bold'),
      axis.ticks.x = element_line())
```

Some quick analysis of read data:
```{r}
stats04 <- read.delim('~/Desktop/Thesis/Projects/IPSC_KRAS/04/read_stats.txt', sep = ' ', header = F)
names(stats04)[1] <- '04'
stats01 <- read.delim('~/Desktop/Thesis/Projects/IPSC_KRAS/01/read_stats.txt', sep = ' ', header = F)
names(stats01)[1] <- '01'
stats12 <- read.delim('~/Desktop/Thesis/Projects/IPSC_KRAS/12/read_stats.txt', sep = ' ', header = F)
names(stats12)[1] <- '12'

s4 <- ggplot(stats04) + 
  geom_density(aes(`04`)) + 
  this_theme + ylab('Count') + xlab('Length')

s1 <- ggplot(stats01) +
  geom_bar(aes(`01`)) + 
  this_theme + ylab('Count') + xlab('Length')

ggplot(stats12) + 
  geom_density(aes(`12`)) + 
  this_theme + ylab('Count') + xlab('Length')

te_ipsc <- read.csv('~/Desktop/Thesis/Projects/IPSC_KRAS/TEs/te_out/Stat_test/results.csv')

te_plot <- ggplot(te_ipsc,
                   aes(log2FoldChange, -log10(pvalue), color=clade, label=name, size=6)) + 
  geom_point() +
  geom_vline(xintercept = 0, linetype='dotted') + 
  theme(text = element_text(size=14, family = "sans"),
        legend.title = element_text(size = 10),
        axis.text.x = element_text(color='black', size=10, face = 'bold'),
        axis.text.y = element_text(face = 'bold', color = 'black', size=10),
        panel.background = element_rect(fill ='white'),
        axis.line.x = element_line(size = 1),
        axis.line.y = element_line(size=1),
        strip.background = element_blank(),
        strip.text = element_blank(),
        legend.position='top',
        legend.key = element_blank(),
        legend.text = element_text(face = 'bold', color = 'black', size=7.5),
        axis.ticks.y = element_line(size=1.5, color='black'), 
        axis.title = element_text(face='bold', size=10),
        axis.ticks.x = element_line(size = 1.5, color='black'),
        axis.ticks.length = unit(.25, 'cm')) + 
  geom_hline(yintercept = -log10(0.05), linetype = 'dotted') +
  ggrepel::geom_text_repel(aes(label = ifelse(abs(log2FoldChange) > 0.8 & pvalue < 0.05,
                                              as.character(name), '')),
                           segment.color = 'black', 
                           color='black',
                           segment.alpha = .5,
                           xlim = c(-Inf, Inf), 
                           ylim = c( -Inf, Inf),
                           point.padding = 0.3,
                           size = 4)
```


TPM
```{r}
quant1 <- read.delim('01_quant.sf')[, c(1,4)]
names(quant1)[2] <- 'KD_TPM'
quant4 <- read.delim('04_quant.sf')[, c(1,4)]
names(quant4)[2] <- 'REP_1_TPM'
quant12 <- read.delim('12_quant.sf')[, c(1,4)]
names(quant12)[2] <- 'REP_2_TPM'

quant_merge <- merge(quant1,quant12, by.x = 'Name', by.y = 'Name')
quant_true_merge <- merge(quant4, quant_merge, by.x = 'Name', by.y = 'Name')

write.csv(quant_true_merge, 'directRNA_tpm.csv')

```

Counts:
```{r}
counts <- read.delim('htseq_count_out.txt', header = F)
names(counts) <- c('Gene','01','04','12')
counts <- as.tibble(counts)
#counts$`01`/colSums(count_gene$`01`)
gencode <- read.delim('gencode.v29.names.txt')
count_gene <- merge(counts, gencode, by.x = 'Gene', by.y = 'Geneid')


```

Correlation:
```{r}
c1 <- ggplot(count_gene) + 
  geom_smooth(aes(`01`, `04`), method = 'lm') + 
  geom_point(aes(`01`, `04`)) +
  this_theme
c2 <- ggplot(count_gene) + 
  geom_smooth(aes(`01`, `12`), method = 'lm') +
  geom_point(aes(`01`, `12`)) + 
  this_theme
c3 <- ggplot(count_gene) + 
  geom_smooth(aes(`12`, `04`), method = 'lm') +
  geom_point(aes(`12`, `04`)) +
  this_theme
c4 <- ggplot(count_gene) + 
  geom_smooth(aes(`01`, `01`), method = 'lm') +
  geom_point(aes(`01`, `01`)) +
  this_theme

cAll <- ggarrange(c1,c2,c3,c4)

```

Lnc_conservation:
```{r}
lnc_cons <- read.delim('~/Desktop/Thesis/Projects/IPSC_KRAS/lncs/lnc_summary_missing_zeros.txt', header=F)
names(lnc_cons) <- c('Chr','Start',"End", 'Cons','SD')
lnc_cons$Start <- lnc_cons$Start - 1

lnc_bed <- read.delim('~/Desktop/Thesis/Projects/IPSC_KRAS/lncs/lnc_exon.bed', header = F)
names(lnc_bed) <- c('Chr','Start',"End", 'Transcript')

zero_exons <- anti_join(lnc_bed, lnc_cons)
zero_exons$Cons <- 0
zero_exons$SD <- 0
zero_exons$Transcript <- NULL
lnc_cons_final <- rbind(lnc_cons, zero_exons)
lnc_cons_merge <- merge(lnc_cons_final, lnc_bed, by.all = c('Chr','Start','End'))
lnc_cons_merge <- lnc_cons_merge %>% separate(Transcript, sep = '_e', into = c('ENST','junk'))
lnc_cons_merge$junk <- NULL

lnc_express_merge <- merge(lnc_cons_merge, consensus_lnc, by.all = 'ENST')
lnc_cons_express <- lnc_express_merge %>% dplyr::group_by(GENE) %>% summarise(b = mean(b), cons = mean(Cons), sd = mean(SD))



lnc_cons_plot <- ggplot(lnc_cons_express) + 
  geom_point(aes(cons, b)) + 
  geom_smooth(aes(cons, b), method = 'lm') +
  this_theme

```

```{r}
cons <- read.delim('~/Desktop/Thesis/Projects/IPSC_KRAS/lncs/all_exon_summary', header=F)
names(cons) <- c('Chr','Start',"End", 'Cons','SD')
cons$Start <- cons$Start - 1

bed <- read.delim('~/Desktop/Thesis/Projects/IPSC_KRAS/lncs/all_exon.bed', header = F)
names(bed) <- c('Chr','Start',"End", 'Transcript')

all_zero_exons <- anti_join(lnc_bed, lnc_cons)
all_zero_exons$Cons <- 0
all_zero_exons$SD <- 0
all_zero_exons$Transcript <- NULL
all_cons_final <- rbind(cons, all_zero_exons)
all_cons_merge <- merge(all_cons_final, bed, by.all = c('Chr','Start','End'))
all_cons_merge <- all_cons_merge %>% separate(Transcript, sep = '_e', into = c('ENST','junk'))
all_cons_merge$junk <- NULL

all_express_merge <- merge(all_cons_merge, consensus, by.all = 'ENST')
all_cons_express <- all_express_merge %>% dplyr::group_by(GENE) %>% summarise(b = mean(b), cons = mean(Cons), sd = mean(SD))


subset(all_cons_express, GENE %in% hervCons$GENE)
all_cons_plot <- ggplot(all_cons_express) + 
  geom_point(aes(cons, b, color = ifelse(GENE %in% hervCons$GENE, 'HERV_LNC', 'Other'))) + 
  geom_smooth(aes(cons, b), method = 'lm') +
  this_theme + 
  scale_color_tron(guide = F, name ='' )
```

